### PR TITLE
Add schema.org structured data to homepage and product details

### LINKED ISSUE #297: Add Schema.org markup
Schema.org markup is used to describe the page content for search engines. 
We should implement it on the product, category/collection, search, and main page.
The current implementation in the old Saleor can be taken as an inspiration

### PR DESCRIPTION
**Pull request by koradon**
I want to merge this change because...

fixes #297 

This can be tested using https://search.google.com/structured-data/testing-tool/u/0/

TODO:
- [ ] Add brand to product structured data

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: src/core/SEO/Homepage/structuredData.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { searchUrl } from "../../../components/App/routes"
+
+export const structuredData = (shop) => {
```
Comment (piotrgrundas) [inline src/core/SEO/Homepage/structuredData.tsx]: Why let? 
IMO just directly return  `JSON.stringify({ (...) })`

File: src/core/SEO/Homepage/structuredData.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    let data = {
```
Comment (piotrgrundas) [inline src/core/SEO/Homepage/structuredData.tsx]: Why let?
IMO just directly return JSON.stringify({ ... })

Comment (koradon) [reply src/core/SEO/Homepage/structuredData.tsx]: This is still WIP sorry for missing label

File: src/userAccount/views/OrderDetails/types/OrderById.ts [hunk 3]
```diff
@@ -215,7 +215,7 @@ export interface OrderById_order {
   number: string | null;
   shippingAddress: OrderById_order_shippingAddress | null;
   /**
-   * List of order lines for the order
+   * List of order lines.
```
Comment (NyanKiyoshi) [inline src/userAccount/views/OrderDetails/types/OrderById.ts]:
```suggestion
   * List of the order lines.
```

File: src/userAccount/views/OrderDetails/types/OrderByToken.ts [hunk 4]
```diff
@@ -215,7 +215,7 @@ export interface OrderByToken_orderByToken {
   number: string | null;
   shippingAddress: OrderByToken_orderByToken_shippingAddress | null;
   /**
-   * List of order lines for the order
+   * List of order lines.
```
Comment (NyanKiyoshi) [inline src/userAccount/views/OrderDetails/types/OrderByToken.ts]:
```suggestion
   * List of the order lines.
```

File: src/userAccount/views/OrderDetails/types/OrderDetail.ts [hunk 5]
```diff
@@ -215,7 +215,7 @@ export interface OrderDetail {
   number: string | null;
   shippingAddress: OrderDetail_shippingAddress | null;
   /**
-   * List of order lines for the order
+   * List of order lines.
```
Comment (NyanKiyoshi) [inline src/userAccount/views/OrderDetails/types/OrderDetail.ts]:
```suggestion
   * List of the order lines.
```

File: src/views/Category/types/Category.ts [hunk 6]
```diff
@@ -68,8 +68,7 @@ export interface Category_products_edges_node {
    */
   thumbnail2x: Category_products_edges_node_thumbnail2x | null;
   /**
-   * The product's base price (without any discounts
-   *         applied).
+   * The product's default base price.
```
Comment (NyanKiyoshi) [inline src/views/Category/types/Category.ts]:
```suggestion
   * The product's default price.
```

File: src/views/Collection/types/Collection.ts [hunk 7]
```diff
@@ -89,8 +89,7 @@ export interface Collection_products_edges_node {
    */
   thumbnail2x: Collection_products_edges_node_thumbnail2x | null;
   /**
-   * The product's base price (without any discounts
-   *         applied).
+   * The product's default base price.
```
Comment (NyanKiyoshi) [inline src/views/Collection/types/Collection.ts]:
```suggestion
   * The product's default price.
```

File: src/views/Product/types/ProductDetails.ts [hunk 8]
```diff
@@ -87,8 +87,7 @@ export interface ProductDetails_product_category_products_edges_node {
   thumbnail2x: ProductDetails_product_category_products_edges_node_thumbnail2x | null;
   category: ProductDetails_product_category_products_edges_node_category;
   /**
-   * The product's base price (without any discounts
-   *         applied).
+   * The product's default base price.
```
Comment (NyanKiyoshi) [inline src/views/Product/types/ProductDetails.ts]:
```suggestion
   * The product's default price.
```

File: src/views/Product/types/ProductDetails.ts [hunk 9]
```diff
@@ -263,7 +269,7 @@ export interface ProductDetails_product {
   seoTitle: string | null;
   /**
    * Informs about product's availability in the
```
Comment (NyanKiyoshi) [inline src/views/Product/types/ProductDetails.ts]:
```suggestion
   * Informs about the product's availability in the
```

File: src/views/Search/types/SearchProducts.ts [hunk 10]
```diff
@@ -69,8 +69,7 @@ export interface SearchProducts_products_edges_node {
   thumbnail2x: SearchProducts_products_edges_node_thumbnail2x | null;
   category: SearchProducts_products_edges_node_category;
   /**
-   * The product's base price (without any discounts
-   *         applied).
+   * The product's default base price.
```
Comment (NyanKiyoshi) [inline src/views/Search/types/SearchProducts.ts]:
```suggestion
   * The product's default price.
```

**Comment by koradon** [comment]
@NyanKiyoshi your suggestion should be made in saleor because they are autogenerated from graphql schema (they are fields descriptions).

**Comment by koradon** [comment]
`Brand` metadata should be added later.

File: src/core/SEO/Product/structuredData.tsx [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  const variants = product.variants;
+
+  return JSON.stringify({
+    "@context": "https://schema.org/",
+    "@type": "Product",
+    description: !product.seoDescription
+      ? `${product.description}`
+      : `${product.seoDescription}`,
+    image: images,
+    name: !product.seoTitle ? `${product.name}` : `${product.seoTitle}`,
```
Comment (piotrgrundas) [inline src/core/SEO/Product/structuredData.tsx]: No need to wrap it in template strings.
IMO just do:
`maybe(() => product.seoTitle, product.name)`

Same applies above.

File: src/core/SEO/Homepage/structuredData.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+export const structuredData = shop => {
+  return JSON.stringify({
+    "@context": "https://schema.org",
+    "@type": "WebSite",
+    description: shop.description,
+    name: shop.name,
+    potentialAction: {
+      "@type": "SearchAction",
+      "query-input": "required name=q",
+      target: searchUrl + "?q={q}"
```
Comment (piotrgrundas) [inline src/core/SEO/Homepage/structuredData.tsx]: Here U could apply template string

File: src/core/SEO/Product/structuredData.tsx (hunk 11 above)
Comment (koradon) [reply src/core/SEO/Product/structuredData.tsx]: Will this handle a situation when seoTitle is an empty string?
From my testing, it adds empty string when 
```
product.seoTitle = ""
product.name = "some name"
```

Comment (piotrgrundas) [reply src/core/SEO/Product/structuredData.tsx]: In that case ur implementation is correct.

**Pull request merged**